environment.yml

```diff
@@ -4,26 +4,28 @@
 name: nupic.research
 channels:
   - defaults
+  - pytorch
+  - nvidia
+  - conda-forge
 
 dependencies:
-  - python=3.7
-  - pip=20.2
+  - python=3.8
+  - pip
 
   # FIXME: remove numba installation command once dependency issues with numpy 1.21.2 is fixed on a new numba release
   #        Caused by this PR https://github.com/numba/numba/pull/7209
   - numba=0.53.0
-
-  # Dependencies not available in conda
   - pip:
-      # Install nupic.torch and  nupic.tensorflow libraries in develop mode.
-      # Assume both libraries were cloned into ${HOME}/nta folder
-      # If you don't want to install nupic.torch and nupic.tensorflow in develop
-      # mode just comment the following lines and use the versions indicated in
-      # the requirements.txt file
-      - -e ${HOME}/nta/nupic.torch --no-binary :all
-      - -e ${HOME}/nta/nupic.tensorflow --no-binary :all
+      # Install nupic.torch libraries in develop mode.
+      # Assume libraries were cloned into ${HOME}/nta folder
+      - -e ${HOME}/nta/nupic.torch
+
+      # Install nupic.research in development mode with dev tools extras enabled
+      - -e ".[dev]"
 
-      # Install nupic.research in develop mode
-      - -r requirements-dev.txt
-      - -e . --no-binary :all
+      # Install projects extra dependency
+      # See `setup.cfg` for available extras.
+      - -e ".[vernon,imagenet,gsc,dendrites]"
```

Comment:
Is the implication that imagenet and gsc are installed by default when installing nupic.research? Should we make this assumption? 

I would argue moving forward we are working a lot more with dendrites, transformers and embodied. imagenet and gsc are more legacy projects.

Reply:
When you install `nupic.research` without any projects, only the dependencies listed in the [install_requires](https://github.com/numenta/nupic.research/pull/569/files#diff-fa602a8a75dc9dcc92261bac5f533c2a85e34fcceaff63b3a3a81d9acde2fc52R50-R61) section will be installed by default

This file is only relevant when creating a new `conda` environment not the `nupic.research` package itself. 

I am installing the current projects to keep the current `conda` environment compatible with this PR. But I agree with you in the future we should deprecate these projects and remove them from the `conda` environment as well.

Reply:
Another option, is to keep the default `conda` environment clean from any project dependency and let the end user install the projects they use on-demand.